ts/features/idpay/onboarding/xstate/machine.ts

```diff
@@ -285,8 +322,61 @@ const createIDPayOnboardingMachine = () =>
                 target: "DISPLAYING_INITIATIVE"
               }
             ]
+          },
+          onDone: {
+            target: "ACCEPTING_REQUIRED_CRITERIA"
+            // triggered by the 'final' substate
+          },
+          states: {
+            EVALUATING_SELF_CRITERIA: {
+              tags: [LOADING_TAG],
+              always: [
+                {
+                  target: "DISPLAYING_BOOL_CRITERIA",
+                  cond: "hasBoolRequiredCriteria"
+                },
+                {
+                  target: "CYCLING_MULTI_CRITERIA",
+                  cond: "hasMultiRequiredCriteria"
+                }
+              ]
+            },
+            DISPLAYING_BOOL_CRITERIA: {
+              tags: [WAITING_USER_INPUT_TAG],
+              entry: "navigateToBoolSelfDeclarationsScreen",
+              on: {
+                ACCEPT_REQUIRED_BOOL_CRITERIA: [
+                  {
+                    target: "CYCLING_MULTI_CRITERIA",
+                    cond: "hasMultiRequiredCriteria"
+                  },
+                  {
+                    target: "MULTI_PREREQUISITES_CYCLE_OVER"
+                  }
+                ]
```

Comment:
```suggestion
                ],
                GO_BACK: [
              {
                target: "#IDPAY_ONBOARDING.DISPLAYING_REQUIRED_PDND_CRITERIA",
                cond: "hasPDNDRequiredCriteria"
              },
              {
                target: "#IDPAY_ONBOARDING.DISPLAYING_INITIATIVE"
              }
            ]
```

The `GO_BACK` event should be placed here and not in the parent state, this will avoid any `GO_BACK` event in the avaluation/loading state